include/flatbuffers/stl_emulation.h

```diff
@@ -20,11 +20,13 @@
 // clang-format off
 #include "flatbuffers/base.h"
 
+#ifndef FLATBUFFERS_NO_STL
```

Comment:
Updated to remove any STL usage from flatbuffers.h. From our perspective, limited STL usage, like std::is_scalar or std::integral_constant, that are easy to support should be fine to use. But it's hard to determine the appropriate macro to guard those with since LIMITED_STL is too bland to really know what to put behind it. Doing true NO_STL puts a decent amount behind the macro. Lemme know what you think with the latest patchset and if this is now too much to keep track of.